src/services/TxHistory/helpers.tsx

```diff
@@ -0,0 +1,63 @@
+import { bigNumberify, parseEther } from 'ethers/utils';
+
+import { ITxHistoryApiResponse } from '@services/ApiService/History';
+import { ITxReceipt, Network, Asset, StoreAccount } from '@types';
+import { getAssetByContractAndNetwork, getBaseAssetByNetwork } from '@services';
+import { ITxHistoryType } from '@features/Dashboard/types';
+import { isSameAddress } from '@utils';
+
+export const makeTxReceipt = (
+  tx: ITxHistoryApiResponse,
+  network: Network,
+  assets: Asset[]
+): ITxReceipt => {
+  const contractAsset = getAssetByContractAndNetwork(tx.to, network)(assets);
+  const baseAsset = getBaseAssetByNetwork({
+    network,
+    assets
+  });
+
+  return {
+    ...tx,
+    asset: contractAsset || baseAsset!,
+    baseAsset: baseAsset!,
+    receiverAddress: tx.recipientAddress,
+    amount: tx.value.toString(),
+    data: tx.data,
+    gasPrice: bigNumberify(tx.gasPrice),
+    gasLimit: bigNumberify(tx.gasLimit),
+    gasUsed: bigNumberify(tx.gasUsed || 0),
+    value: parseEther(tx.value.toString()),
+    nonce: tx.nonce.toString()
+  };
+};
+
+export const merge = (apiTxs: ITxReceipt[], accountTxs: ITxReceipt[]): ITxReceipt[] => {
+  // Prioritize Account TX - needs to be more advanced?
+  const filteredApiTxs = apiTxs.filter((tx) => !accountTxs.find((a) => a.hash === tx.hash));
+  return filteredApiTxs.concat(accountTxs).sort((a, b) => (a.timestamp || 0) - (b.timestamp || 0));
```

Comment:
I might be mistaken, but I believe the base table component already sorts the txs when a sortable column is specified. Do we need them to be available pre-sorted in the context for some other reason?

Reply:
No, you are right! 👍